app/Http/Controllers/SquareCheckoutController.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+declare(strict_types=1);
+
+// phpcs:disable Generic.Formatting.SpaceBeforeCast.NoSpace
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\SquareCompleteRequest;
+use App\Models\DuesTransaction;
+use App\Models\Payment;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Square\Models\CreateCheckoutRequest;
+use Square\Models\CreateOrderRequest;
+use Square\Models\Money;
+use Square\Models\Order;
+use Square\Models\OrderLineItem;
+use Square\Models\OrderServiceCharge;
+use Square\Models\OrderServiceChargeCalculationPhase;
+use Square\Models\OrderState;
+use Square\SquareClient;
+
+class SquareCheckoutController extends Controller
+{
+    private const PER_TRANSACTION_FEE = 30;  // cents
+    private const PERCENTAGE_FEE = 2.9;
+
+    public function payDues(Request $request)
+    {
+        $user = $request->user();
+
+        $transactionWithNoPayment = DuesTransaction::doesntHave('payment')
+            ->where('user_id', $user->id)
+            ->latest('updated_at')
+            ->first();
+
+        $transactionWithIncompletePayment = DuesTransaction::where('user_id', $user->id)
+            ->whereHas('package', static function (Builder $q): void {
+                $q->whereDate('effective_end', '>=', date('Y-m-d'));
```

Comment:
Do we have an active() local scope that could be used here instead?

Reply:
Yes. Would that just be...

```diff
 $transactionWithIncompletePayment = DuesTransaction::where('user_id', $user->id)
-    ->whereHas('package', static function (Builder $q): void {
-        $q->whereDate('effective_end', '>=', date('Y-m-d'));
-    })->whereHas('payment', static function (Builder $q): void {
+    ->current()
+    ->whereHas('payment', static function (Builder $q): void {
         $q->where('amount', 0.00);
         $q->where('method', 'square');
     })->first();
```

Reply:
Yes, that looks correct. Note that the DuesPackage active() scope checks both effective start and end.